addons/pkg/util/addon_util.go

```diff
@@ -217,6 +218,23 @@ func GetPackageInstallFromAddonSecret(ctx context.Context,
 	return pkgi, nil
 }
 
+// DeletePackageInstall deletes the PackageInstall CR
+func DeletePackageInstall(ctx context.Context, c client.Client, pkgiName, pkgiNamespace string) error {
+	pkgi := &pkgiv1alpha1.PackageInstall{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      pkgiName,
+			Namespace: pkgiNamespace,
+		},
+	}
+	pkgi.Spec.NoopDelete = true
+
```

Comment:
Should we update the package install before deleting it? 
i.e. here the nooopdelete is set in the pkgi in memory. 
Then you call delete imeditaley. Not sure if the noopdelete ever makes it to the actual database. It could be that the Delete just looks at the name and namespace. 
I think we should patch the object with noopdelete first, then delete it.

Reply:
need to do a 
key = metava1.objectkey(name: pkginma, namespace: pkginamespace)
error := c.get(ctx, key, pkgi)
pkgi.spec.noopdelete
then patch
then call tkgclient.packageuninstall.

Reply:
moved to management components under pkg/v1/tkg